### PR TITLE
Nexus Docker Apt Config

### PR DESCRIPTION
**Pull request by oliver7598**
# PR for issue #1274

## What is being addressed
Docker should be able to be installed on a Linux VM as detailed [here](https://docs.docker.com/engine/install/ubuntu/)

## How is this addressed
- Additional json config file added to configure Nexus to allow the download of the public gpg key at :
https://nexus-<tre-id>.azurewebsites.net/repository/docker-public-key/gpg
- Additional json config file added to configure Nexus to provide a dock proxy apt repo at https://nexus-<tre-id>.azurewebsites.net/repository/docker
- VM Deploy script had additional step to download gpg key and store at `/etc/apt/trusted.gpg.d/docker-archive-keyring.gpg`
- Apt source added signed by gpg key for https://nexus-<tre-id>.azurewebsites.net/repository/docker

## Additional Changes to Nexus Configuration
There were also a few additional changes to streamline the addition of future Nexus configurations.
- All json config files moved from `.\scripts\` to `.\scripts\nexus_config\`
- Two additional properties added to all json config files. `baseType` and `repoType`.  `baseType` is the type of proxy being configured (e,g, apt, docker, r, nuget, raw, pypi) and  `repoType` is the type of repo being set up (proxy, hosted, group)
- configure_nexus.sh now reads the `.\scripts\nexus_config\` folder and for each json file in the folder the repository will be created and configure.
- Addition of future nexus repositories will now only require a new json file to be added.

## Additional Changes to Linux `vm_config.sh` file
- Content of `/etc/apt/sources.list.d/` folder has been cleared.  On running apt-get update apt sources in this folder were used to update existing packages resulting in a series of errors being displayed in the console (See example below)
image

NOTE: I have also left in the creation of the Nexus Docker Hub proxy and the opening of the firewall to allow communication to `*.docker.com` and `*.docker.io` as a basis for addressing #1089.  Although, as mentioned in the comments in the issue, further port configuration is required to fully achieve this and there are ongoing discussions on how this will be achieved.

### CODE REVIEW

**Review by marrobi — APPROVED** [review]
LGTM

File: scripts/configure_nexus.sh [hunk 1]
```diff
@@ -62,67 +62,26 @@ if [ -z "$NEXUS_PASS" ]; then
     export NEXUS_PASS=$NEW_PASSWORD
```
Comment (tamirkamara) [inline scripts/configure_nexus.sh:62]: Not the perfect place to put this comment in:
Seems like we have a lot of files directly related to Nexus at the root of the project folder structure. What do you think about moving all of them into the nexus template folder @oliver7598 ?

**Review by tamirkamara — CHANGES_REQUESTED** [review]

File: scripts/configure_nexus.sh (hunk 1 above)
Comment (marrobi) [reply scripts/configure_nexus.sh:62]: I tend to agree. Although might be that's taken as a separate issue that includes building these steps into the nexus bundle?

Comment (tamirkamara) [reply scripts/configure_nexus.sh:62]: I'm fine with waiting as long as we open an issue and reference it here for the comment to be resolved. 
Although, I don't see any knock-on affect with regards to the nexus things (aad app reg is outside the scope). If this is going to impact github actions and/or AzDo pipelines something else is wrong as we shouldn't directly use any of this...

To clarify, we're talking about moving the configure_nexus.sh script and the nexus_config folder into the template folder.

Comment (oliver7598) [reply scripts/configure_nexus.sh]: Nexus config scripts and related files have been moved into the nexus shared-service folder.  

Docs also updated to reflect new script path in configure shared services section

**Review by tamirkamara — APPROVED** [review]
LGTM. Thank you for contributing.

**Pull request merged**